### PR TITLE
Initial cut of wallet layout EIP.

### PR DESCRIPTION
**Pull request by mcdee**
Initial wallet layout proposal.

### CODE REVIEW

**Comment by MicahZoltu** [comment]
I don't know why CI is failing for a bunch of unrelated PRs, but once we figure that out I'm good to merge this.

**Comment by MicahZoltu** [comment]
Kicking CI.

File: EIPS/eip-2680.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+Putting these elements together for a sample wallet with wallet UUID `1f031fff-c51d-44fc-8baf-d6b304cb70a7` and key UUIDs `1302106c-8441-4e2e-b687-6c77f49fc624` and `4a320100-83fd-4db7-8126-6d6d205ba834` gives the following layout:
+
+```
+- 1f031fff-c51d-44fc-8baf-d6b304cb70a7
++- 1302106c-8441-4e2e-b687-6c77f49fc624
++- 1f031fff-c51d-44fc-8baf-d6b304cb70a7
++- 4a320100-83fd-4db7-8126-6d6d205ba834
+```
+
+## Non-hierarchical store example
```
Comment (MicahZoltu) [inline EIPS/eip-2680.md:68]: This and the following sections should be in one of the template sections.  My guess is you just forgot a `#` at the start?
```suggestion
### Non-hierarchical store example
```

File: EIPS/eip-2680.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+Non-hierarchical stores use a simplified approach where the wallet UUID and key UUIDs are concatenated using the ':' character.  Using the same example wallet and key UUIDs as above would result in objects with the following keys:
+
+```
+1f031fff-c51d-44fc-8baf-d6b304cb70a7:1302106c-8441-4e2e-b687-6c77f49fc624
+1f031fff-c51d-44fc-8baf-d6b304cb70a7:1f031fff-c51d-44fc-8baf-d6b304cb70a7
+1f031fff-c51d-44fc-8baf-d6b304cb70a7:4a320100-83fd-4db7-8126-6d6d205ba834
+```
+
+## Protecting against concurrent write access
```
Comment (MicahZoltu) [inline EIPS/eip-2680.md:77]:
```suggestion
### Protecting against concurrent write access
```

File: EIPS/eip-2680.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+TBD
+
+## Iterating over wallets
```
Comment (MicahZoltu) [inline EIPS/eip-2680.md:80]:
```suggestion
### Iterating over wallets
```

File: EIPS/eip-2680.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+In the case of hierarchical stores and iteration-capable non-hierarchical stores iteration over wallets is a matter of iterating over the files in the root container.
+
+An implementer MAY include an index in the base location.  If so then it MUST follow the structure as specified in the following "Index format" section.
+
+## Iterating over accounts
```
Comment (MicahZoltu) [inline EIPS/eip-2680.md:85]:
```suggestion
### Iterating over accounts
```

File: EIPS/eip-2680.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+In the case of hierarchical stores iteration over accounts is a matter of iterating over the files in the wallet container.
+
+An implementer MAY include an index within a wallet container for accounts within that wallet.  If so then it MUST follow the structure as specified in the following "Index format" section.
+
+## Index format.
```
Comment (MicahZoltu) [inline EIPS/eip-2680.md:90]:
```suggestion
### Index format
```

File: EIPS/eip-2680.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+A standard for walletstores, similar to that for keystores, provides a higher level of compatibility between wallets and allows for simpler wallet and key interchange between them.
+
+## Implementation
+
+A Go implementation of the filesystem layout can be found at [https://github.com/wealdtech/go-eth2-wallet-filesystem](https://github.com/wealdtech/go-eth2-wallet-filesystem).
+
+A Go implementation of the Amazon S3 layout can be found at [https://github.com/wealdtech/go-eth2-wallet-s3](https://github.com/wealdtech/go-eth2-wallet-s3).
+
+## Copyright
```
Comment (MicahZoltu) [inline EIPS/eip-2680.md:127]: The `Security Considerations` section is missing, and is required.
```suggestion
## Security Considerations
TBD

## Copyright
```

File: EIPS/eip-2680.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+            "name": {
+                "type": "string"
+            }
+        },
+        "required": [
+            "uuid",
+            "name"
+        ]
+    }
+}
```
Comment (MicahZoltu) [inline EIPS/eip-2680.md:111]:
````suggestion
}
```
````
Missing code block close ticks.

**Review by MicahZoltu — CHANGES_REQUESTED** [review]
One last (hopefully) item to get this merged.

**Comment by MicahZoltu** [comment]
We are trying to get rid of external links from EIPs as much as possible.  While it won't block merging as draft, it *may* cause problems down the road when merging as Last Call or Final.  Can you look into inlining the content of the external links into this EIP, or adding them as assets in `../assets/eip-2680/`?

**Comment by MicahZoltu** [comment]
Sorry, I missed that this one wasn't addressed yet: https://github.com/ethereum/EIPs/pull/2680#discussion_r456768459  Also required for merge.

**Comment by mcdee** [comment]
Updated one of the external links with an internal asset.  Wanted to confirm this is what is desired, as I can't see any non-image assets elsewhere.  Also, if this is correct do we _really_ want to take local copies of RFCs?

**Comment by MicahZoltu** [comment]
Yeah, that is what I meant.

There exists a spectrum between "link to an RFC" and "link to page that is obviously marketing material meant to pump a shitcoin and will likely 404 by the end of the year".  EIP Editors have a long history of taking no political stance and picking hard line rules that don't require judgement to enforce, and this is one of those stances.  If we let RFCs through, why not IEEE, why not W3C, why not BIPs, why not ShitcoinIPs, etc.  It is easier for us to just have a blanket rule for no external links than to try to enforce a fuzzy line.

To be clear, this isn't a hard and fast rule and the editors may decide to let RFC links through, but the process has less chance of contention if you inline the assets.

**Pull request merged**